plutus-pab/src/Plutus/PAB/Db/Beam/ContractStore.hs

```diff
@@ -104,8 +118,17 @@ handleContractStore = \case
         extractState = \case
           Nothing -> throwError $ ContractInstanceNotFound instanceId
           Just  c ->
-            fromMaybe (throwError $ ContractStateNotFound instanceId)
-                      (pure <$> (_contractInstanceState c >>= decodeText))
+            do
+              caID <- maybe (throwError $ AesonDecodingError "Couldn't this JSON decode as `a`" (_contractInstanceId c))
```

Comment:
We could use [`typeRep`](https://hackage.haskell.org/package/base-4.10.1.0/docs/Data-Typeable.html#v:typeRep) to get the name of the type for a more user-friendly error message